// @actonate/web-rx/dist/auth/FacebookLogin.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _reactFacebookLogin = _interopRequireDefault(require("react-facebook-login"));

var _styledSystem = require("styled-system");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FacebookLoginRx = function FacebookLoginRx(props) {
  return _react.default.createElement(_reactFacebookLogin.default, {
    cssClass: `web-rx facebook-login icon ${props.className}`,
    appId: props.appId,
    autoLoad: false,
    fields: props.fields,
    scope: props.scope,
    callback: props.onSubmit,
    textButton: props.label,
    onFailure: props.onError
  });
};

FacebookLoginRx.defaultProps = {
  className: null,
  fields: 'name,email,picture',
  scope: 'public_profile',
  label: 'Continue with Facebook',
  onSubmit: function onSubmit() {},
  onError: function onError() {}
};
FacebookLoginRx.propTypes = {
  className: _propTypes.default.string,
  appId: _propTypes.default.string.isRequired,
  fields: _propTypes.default.string,
  scope: _propTypes.default.string,
  label: _propTypes.default.string,
  onSubmit: _propTypes.default.func,
  onError: _propTypes.default.func
};
var StyledFacebookLogin = (0, _styledComponents.default)(FacebookLoginRx)`
  display: flex;
  width: 100%;
  margin-right: auto;
  margin-bottom: 15px;
  margin-left: auto;
  padding: 10px 10px 10px 42px;
  align-items: center;
  border-radius: 5px;
  background-color: #3b5997;
  box-shadow: 1px 0 2px 0 rgba(0, 0, 0, .16);
  color: #fff;
  text-decoration: none;
  justify-content: flex-start;
  outline: 0;
  &:hover, &:active {
    outline: 0;
  }
  &.icon {
    background-size: 21px;
    background-repeat: no-repeat;
    background-position: 10px center;
  }
  ${_styledSystem.space}
  ${_styledSystem.height}
  ${_styledSystem.width}
`;
var _default = StyledFacebookLogin;
exports.default = _default;
