// @actonate/web-rx/dist/auth/GoogleLogin.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _reactGoogleLogin = _interopRequireDefault(require("react-google-login"));

var _styledSystem = require("styled-system");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var GoogleLoginRx = function GoogleLoginRx(props) {
  return _react.default.createElement(_reactGoogleLogin.default, {
    className: `web-rx google-login icon ${props.className}`,
    clientId: props.clientId,
    scope: props.scope,
    buttonText: props.label,
    onSuccess: props.onSubmit,
    onFailure: props.onError
  });
};

GoogleLoginRx.defaultProps = {
  className: null,
  scope: 'profile email',
  label: 'Continue with Google',
  onSubmit: function onSubmit() {},
  onError: function onError() {}
};
GoogleLoginRx.propTypes = {
  className: _propTypes.default.string,
  clientId: _propTypes.default.string.isRequired,
  scope: _propTypes.default.string,
  label: _propTypes.default.string,
  onSubmit: _propTypes.default.func,
  onError: _propTypes.default.func
};
var StyledGoogleLogin = (0, _styledComponents.default)(GoogleLoginRx)`
  display: flex;
  width: 100%;
  margin-right: auto;
  margin-bottom: 15px;
  margin-left: auto;
  padding: 10px 10px 10px 42px;
  align-items: center;
  border-radius: 5px;
  background-color: #db4d41;
  box-shadow: 1px 0 2px 0 rgba(0, 0, 0, .16);
  color: #fff;
  text-decoration: none;
  justify-content: flex-start;
  outline: 0;
  &:hover, &:active {
    outline: 0;
  }
  &.icon {
    background-size: 21px;
    background-repeat: no-repeat;
    background-position: 10px center;
  }
  &:hover {
    cursor: pointer;
  }
  ${_styledSystem.space}
  ${_styledSystem.height}
  ${_styledSystem.width}
`;
var _default = StyledGoogleLogin;
exports.default = _default;
