// @actonate/web-rx/dist/payments/CreditCardNumber.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

require("core-js/modules/es6.object.assign");

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _Icon = _interopRequireDefault(require("semantic-ui-react/dist/commonjs/elements/Icon/Icon"));

var _reactCreditCardPrimitives = require("react-credit-card-primitives");

var _controls = require("../controls");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (typeof call === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

var CreditCardNumber =
/*#__PURE__*/
function (_React$Component) {
  _inherits(CreditCardNumber, _React$Component);

  function CreditCardNumber(props) {
    var _this;

    _classCallCheck(this, CreditCardNumber);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(CreditCardNumber).call(this, props));
    _this.state = {
      value: props.value,
      icon: false
    };
    _this.onChange = _this.onChange.bind(_assertThisInitialized(_assertThisInitialized(_this)));
    return _this;
  }

  _createClass(CreditCardNumber, [{
    key: "onChange",
    value: function onChange(_ref) {
      var value = _ref.value,
          valid = _ref.valid,
          type = _ref.type;
      var icon = false;

      switch (type) {
        case 'Visa':
          icon = _react.default.createElement(_Icon.default, {
            color: "blue",
            size: "large",
            name: "cc visa"
          });
          break;

        case 'Mastercard':
          icon = _react.default.createElement(_Icon.default, {
            color: "red",
            size: "large",
            name: "cc mastercard"
          });
          break;

        case 'Maestro':
          icon = _react.default.createElement(_Icon.default, {
            color: "red",
            size: "large",
            name: "cc maestro"
          });
          break;

        case 'American Express':
          icon = _react.default.createElement(_Icon.default, {
            color: "blue",
            size: "large",
            name: "cc amex"
          });
          break;

        case 'JCB':
          icon = _react.default.createElement(_Icon.default, {
            color: "yellow",
            size: "large",
            name: "cc jcb"
          });
          break;

        case 'Diner':
          icon = _react.default.createElement(_Icon.default, {
            color: "blue",
            size: "large",
            name: "cc diner"
          });
          break;

        case 'Discover':
          icon = _react.default.createElement(_Icon.default, {
            color: "orange",
            size: "large",
            name: "cc discover"
          });
          break;

        default:
          icon = false;
          break;
      }

      this.setState({
        value: value,
        icon: icon
      });
      this.props.onChange(value, valid, type);
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      return _react.default.createElement(_reactCreditCardPrimitives.Number, {
        masked: this.props.masked,
        cardTypes: this.props.cardTypes,
        value: this.state.value,
        onChange: this.onChange,
        render: function render(_ref2) {
          var getInputProps = _ref2.getInputProps,
              valid = _ref2.valid;
          return _react.default.createElement(_controls.Input, _extends({}, getInputProps(), {
            fluid: true,
            icon: _this2.state.icon
          }));
        }
      });
    }
  }]);

  return CreditCardNumber;
}(_react.default.Component);

CreditCardNumber.defaultProps = {
  value: '',
  masked: true,
  cardTypes: ['Visa', 'Mastercard', 'Maestro', 'American Express'],
  onChange: function onChange() {}
};
CreditCardNumber.propTypes = {
  value: _propTypes.default.string,
  masked: _propTypes.default.bool,
  cardTypes: _propTypes.default.arrayOf(_propTypes.default.string),
  onChange: _propTypes.default.func
};
var StyledCreditCardNumber = (0, _styledComponents.default)(CreditCardNumber)`
  ${_styledSystem.space}
  ${_styledSystem.height}
  ${_styledSystem.width}
`;
var _default = StyledCreditCardNumber;
exports.default = _default;
