// @actonate/web-rx/dist/controls/List.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _templateObject = _taggedTemplateLiteral(['\n  ', '\n  ', '\n  ', '\n'], ['\n  ', '\n  ', '\n  ', '\n']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _styledSystem = require('styled-system');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var List = function List(props) {
  return _react2.default.createElement(
    'div',
    { className: 'web-rx list ' + props.className },
    _react2.default.createElement(
      'p',
      null,
      'test'
    )
  );
};

List.defaultProps = {
  items: [],
  className: null,
  select: false,
  multiple: false,
  template: null,
  onChange: function onChange() {}
};

List.propTypes = {
  className: _propTypes2.default.string,
  items: _propTypes2.default.array, // eslint-disable-line
  select: _propTypes2.default.bool,
  multiple: _propTypes2.default.bool,
  onChange: _propTypes2.default.func
};

var StyledList = (0, _styledComponents2.default)(List)(_templateObject, _styledSystem.space, _styledSystem.height, _styledSystem.width);

exports.default = StyledList;
