// @actonate/web-rx/dist/controls/Range.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

require("core-js/modules/web.dom.iterable");

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _rheostat = _interopRequireDefault(require("rheostat"));

var _styledSystem = require("styled-system");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Range = function Range(props) {
  return _react.default.createElement("div", {
    className: `web-rx range ${props.className}`
  }, _react.default.createElement(_rheostat.default, {
    min: props.min,
    max: props.max,
    values: props.values,
    snap: props.snap,
    snapPoints: props.snapPoints,
    onChange: function onChange(event) {
      props.onChange(event.values);
    }
  }));
};

Range.defaultProps = {
  className: null,
  min: 1,
  max: 100,
  snap: false,
  snapPoints: [],
  values: [1, 100],
  onChange: function onChange() {}
};
Range.propTypes = {
  className: _propTypes.default.string,
  min: _propTypes.default.number,
  max: _propTypes.default.number,
  snap: _propTypes.default.bool,
  snapPoints: _propTypes.default.arrayOf(_propTypes.default.number),
  values: _propTypes.default.arrayOf(_propTypes.default.number),
  onChange: _propTypes.default.func
};
var DEFAULT_PRIMARY_COLOR = '#2196F3';
var StyledRange = (0, _styledComponents.default)(Range)`
  .rheostat {
    overflow: visible;
  }

  .rheostat-background {
    background-color: #fcfcfc;
    border: 1px solid #d8d8d8;
    position: relative;
  }

  .rheostat-progress {
    background-color: ${function (props) {
  return props.theme.colors.primary || DEFAULT_PRIMARY_COLOR;
}};
    position: absolute;
  }

  .rheostat-handle {
    background-color: #fff;
    border: 1px solid #d8d8d8;
    border-radius: 100%;
    height: 24px;
    outline: none;
    z-index: 2;
    width: 24px;
  }

  .rheostat-horizontal {
    height: 24px;
  }

  .rheostat-horizontal .rheostat-background {
    height: 2px;
    top: 6px;
    width: 100%;
  }

  .rheostat-horizontal .rheostat-progress {
    height: 3px;
    top: 6px;
  }

  .rheostat-horizontal .rheostat-handle {
    margin-left: -12px;
    top: -5px;
  }

  .rheostat-horizontal .rheostat-handle:before,
  .rheostat-horizontal .rheostat-handle:after {
    top: 7px;
    height: 10px;
    width: 1px;
  }

  .rheostat-horizontal .rheostat-handle:before {
    left: 10px;
  }

  .rheostat-horizontal .rheostat-handle:after {
    left: 13px;
  }

  ${_styledSystem.space}
  ${_styledSystem.height}
  ${_styledSystem.width}
`;
var _default = StyledRange;
exports.default = _default;
