// @actonate/web-rx/dist/containers/Infinite.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _reactInfiniteScrollComponent = _interopRequireDefault(require("react-infinite-scroll-component"));

var _styledSystem = require("styled-system");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var InfiniteContainer = function InfiniteContainer(props) {
  return _react.default.createElement("div", {
    className: `web-rx infinite-container ${props.className}`
  }, _react.default.createElement(_reactInfiniteScrollComponent.default, {
    dataLength: props.count,
    next: props.fetch,
    height: props.height,
    hasMore: !props.isCompleted,
    scrollThreshold: props.threshold,
    initialScrollY: props.jumpToY,
    scrollableTarget: props.target,
    loader: props.loader,
    endMessage: props.endMessage,
    onScroll: props.onScroll
  }, props.children));
};

InfiniteContainer.defaultProps = {
  className: null,
  threshold: 0.7,
  jumpToY: 0,
  height: null,
  target: null,
  children: null,
  isCompleted: false,
  loader: _react.default.createElement("h4", null, "Loading ..."),
  endMessage: _react.default.createElement("p", {
    style: {
      textAlign: 'center'
    }
  }, _react.default.createElement("b", null, "Yay! You have seen it all")),
  onScroll: function onScroll() {}
};
InfiniteContainer.propTypes = {
  className: _propTypes.default.string,
  count: _propTypes.default.number.isRequired,
  fetch: _propTypes.default.func.isRequired,
  isCompleted: _propTypes.default.bool,
  threshold: _propTypes.default.number,
  jumpToY: _propTypes.default.number,
  height: _propTypes.default.number,
  target: _propTypes.default.node,
  children: _propTypes.default.node,
  loader: _propTypes.default.node,
  endMessage: _propTypes.default.node,
  onScroll: _propTypes.default.func
};
var StyledInfiniteContainer = (0, _styledComponents.default)(InfiniteContainer)`
  ${_styledSystem.space}
  ${_styledSystem.maxWidth}
  ${_styledSystem.minWidth}
  ${_styledSystem.height}
  ${_styledSystem.width}
`;
var _default = StyledInfiniteContainer;
exports.default = _default;
